gadget/gadget_test.go

```diff
@@ -3754,3 +3755,48 @@ func (s *gadgetYamlTestSuite) TestHasRoleUnhappy(c *C) {
 	_, err = gadget.HasRole(s.dir, []string{gadget.SystemData})
 	c.Check(err, ErrorMatches, `cannot minimally parse gadget metadata: yaml:.*`)
 }
+
+func appendAllowListToYaml(allow []string, templ string) string {
+	for _, arg := range allow {
+		templ += fmt.Sprintf("    - %s\n", arg)
+	}
+	return templ
+}
+
+func (s *gadgetYamlTestSuite) TestKernelCmdlineAllow(c *C) {
+	yamlTemplate := `
+volumes:
+  pc:
+    bootloader: grub
+kernel-cmdline:
+  allow:
+`
+
+	tests := []struct {
+		allowList []string
+		err       string
+	}{
+		{[]string{"foo=bar", "my-param.state=blah"}, ""},
```

Comment:
Should we add a test here that tests `foo=` and rejects it (I assume we need to reject it)?

Reply:
Test added, but this is not rejected by current code, and I think that is fine as the expression is not really malformed, in the end having or not the = does not change what would be matched.